packages/x-date-pickers-pro/src/TimeRangePicker/TimeRangePickerToolbar.tsx

```diff
@@ -0,0 +1,355 @@
+  // ----------------------------------------------------------------------
+  ampm: PropTypes.bool.isRequired,
+  /**
+   * Callback fired on view change.
+   * @template TView
+   * @param {TView} view The new view.
+   */
+  onViewChange: PropTypes.func.isRequired,
+  pickerVariant: PropTypes.oneOf(['desktop', 'mobile']).isRequired,
+  separatorClasses: PropTypes.string.isRequired,
+  /**
+   * Toolbar value placeholder—it is displayed when the value is empty.
+   * @default "––"
+   */
+  toolbarPlaceholder: PropTypes.node,
+  utils: PropTypes.object.isRequired,
+  value: PropTypes.object,
+  /**
+   * The visible view.
+   * Used when the component view is controlled.
+   * Must be a valid option from `views` list.
+   */
+  view: PropTypes.oneOf(['hours', 'meridiem', 'minutes', 'seconds']),
+  /**
+   * Available views.
+   */
+  views: PropTypes.arrayOf(PropTypes.oneOf(['hours', 'meridiem', 'minutes', 'seconds']).isRequired)
+    .isRequired,
+} as any;
+
+const TimeRangePickerToolbar = React.forwardRef(function TimeRangePickerToolbar(
+  inProps: TimeRangePickerToolbarProps,
+  ref: React.Ref<HTMLDivElement>,
+) {
+  const props = useThemeProps({ props: inProps, name: 'MuiTimeRangePickerToolbar' });
+  const utils = useUtils();
+
+  const { className, ampm, toolbarPlaceholder = '--', classes: classesProp, ...other } = props;
+
+  const { value, view, setView, views } = usePickerContext<
+    PickerRangeValue,
+    TimeViewWithMeridiem
+  >();
+  const translations = usePickerTranslations();
+  const ownerState = useToolbarOwnerState();
+  const { rangePosition, setRangePosition } = usePickerRangePositionContext();
+  const classes = useUtilityClasses(classesProp, ownerState);
+
+  const handleStartRangeViewChange = React.useCallback(
+    (newView: TimeViewWithMeridiem) => {
+      if (rangePosition !== 'start') {
+        setRangePosition('start');
+      }
+      setView(newView);
+    },
+    [setRangePosition, setView, rangePosition],
+  );
+
+  const handleEndRangeViewChange = React.useCallback(
+    (newView: TimeViewWithMeridiem) => {
+      if (rangePosition !== 'end') {
+        setRangePosition('end');
+      }
+      setView(newView);
+    },
+    [setRangePosition, setView, rangePosition],
+  );
+
+  if (!view) {
+    return null;
+  }
+
+  return (
+    <TimeRangePickerToolbarRoot
+      {...other}
+      toolbarTitle={translations.timeRangePickerToolbarTitle}
+      className={clsx(className, classes.root)}
+      ownerState={ownerState}
+      ref={ref}
+    >
+      <TimeRangePickerToolbarContainer
+        className={classes.container}
+        pickerVariant={ownerState.pickerVariant}
+      >
+        <TimeRangePickerToolbarTimeElement
+          view={rangePosition === 'start' ? view : undefined}
+          views={views}
+          value={value[0]}
+          onViewChange={handleStartRangeViewChange}
+          ampm={ampm}
+          utils={utils}
```

Comment:
We can use `useUtils` directly in the component :+1: 
Same for anything else accessible through context by the way